File: src/java/tree/226.invert-binary-tree.java
import java.util.Stack;

class P226 {
    // @lc app=leetcode id=226 lang=java
    // @lc code=start
    public class Solution {
        public TreeNode invertTree(TreeNode root) {
            Stack<TreeNode> stack = new Stack<>();
            stack.push(root);
            while(!stack.isEmpty()) {
                TreeNode node = stack.pop();

                if(node != null){
                    TreeNode temp = node.left;
                    node.left = node.right;
                    node.right = temp;
                    stack.push(node.left);
                    stack.push(node.right);
                }
            }
            return root;        
        }
    }
    // @lc code=end
}

File: src/go/tree/226.invert-binary-tree.go
package main

/*
 * @lc app=leetcode id=226 lang=golang
 *
 * [226] Invert Binary Tree
 */

// @lc code=start
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}

	// Swap left and right children
	root.Left, root.Right = root.Right, root.Left

	// Recursively invert subtrees
	invertTree(root.Left)
	invertTree(root.Right)

	return root
}

// @lc code=end
